core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -91,6 +92,13 @@ public static WriteBuilder write(OutputFile file) {
     return new WriteBuilder(file);
   }
 
+  public static WriteBuilder write(EncryptedOutputFile file) {
+    Preconditions.checkState(
+        file.keyMetadata() == null || file.keyMetadata() == EncryptionKeyMetadata.EMPTY,
+        "Currenty, encryption of data files in Avro format is not supported");
```

Comment:
Why is this not supported? The AES GCM stream could easily be used here.

Reply:
We should also run `TestAvroFileSplit` on Avro inside of AES GCM streams

Reply:
SGTM

Reply:
How nice. I expected Avro table encryption to work directly with AES GCM Streams - but not without some hiccups and fixes, since I never ran this usecase before. Turns out it just works out of box. 
Now I have a functioning e2e unitest that encrypts/decrypts an Iceberg table with the Avro data format. The unitest is based on Spark SQL and catalog clients, so will go into the integration PR. 
I'll add an encrypting version of the `TestAvroFileSplit` to this PR.

Reply:
I'll verify of course where the file length comes from for the Avro reader.